### DISCUSSION TITLE
Files that don't contain UI (`.data.ftd` files)

### DISCUSSION THREAD

**Discussion by siddhantk232** [fastn-stack/fastn#2038] [Ideas & RFCs]
I've seen fastn projects with `.ftd` files that do not contain any UI. These files are not meant to be served by the HTTP server but to be used in other modules (`.ftd` files).

Here's the most recent example I found in the `/page-module.ftd` of [website-template](https://fifthtry.com/ide/website-template) (it just contains a record instance):

```ftd
-- ws.about-data about-info: ABOUT
about-img: https://meenukumari.com/-/meenukumari.com/assets/hero-image-dark.svg

Lorem ipsum dolor sit amet consectetur. Tristique amet sed massa nibh lectus netus in. 
Aliquet donec morbi convallis pretium. Turpis tempus pharetra. Lorem ipsum dolor sit amet consectetur. 
Tristique amet sed massa nibh lectus netus in. Aliquet donec morbi convallis pretium. Turpis tempus pharetra.
Lorem ipsum dolor sit amet consectetur. Tristique amet sed massa nibh lectus netus in. 
Aliquet donec morbi convallis pretium. Turpis tempus pharetra.
Lorem ipsum dolor sit amet consectetur. Tristique amet sed massa nibh lectus netus in. 
Aliquet donec morbi convallis pretium. Turpis tempus pharetra.
Lorem ipsum dolor sit amet consectetur. Tristique amet sed massa nibh lectus netus in. 
Aliquet donec morbi convallis pretium. Turpis tempus pharetra.
```

fastn should have a convention (or an enforced rule) that files ending with `.data.ftd` are ignore by the HTTP server.

**Comment by amitu** [comment]
We already support `-- ftd.json:` which short circuits the current package evaluation and returns the JSON. We can create `-- ftd.non-http:` or something, which will effectively make it give a 404. 

If it should be called `ftd.non-http` or we should use `fastn.non-http` no onwards is open.

**Reply by amitu** [reply]
Also you can just use `-- ftd.redirect: /` or something today to achieve the goal of "ignore by the HTTP server". 

So we can return redirect or 404. Making the file name part of convention means if you change your mind you will have to rename, which will break your imports (the whole point of such files is so they are used by other modules via imports).
